draft-ietf-httpbis-priority.md

```diff
@@ -402,10 +402,27 @@ Prioritized Stream ID:
 Priority Field Value:
 : The priority update value in ASCII text, encoded using Structured Fields.
 
-The Prioritized Stream ID MUST be within the stream limit. If a
-server receives a PRIORITY_UPDATE with a Prioritized Stream ID that is beyond
-the stream limits, this SHOULD be treated as a connection error of type
-PROTOCOL_ERROR.
+When the PRIORITY_UPDATE frame applies to a request stream, a client SHOULD
+provide a Prioritized Stream ID that refers to a stream in the "open",
+"half-closed (local)", or "idle" state. Frames where the Prioritized Stream ID
+refers to a stream in the "half-closed (remote)" or "closed" state can be
+discarded by the server upon receipt. The number of streams which have been
+prioritized but remain in the "idle" state plus the number of active streams
+(those in the "open" or "half-closed" states; see section 5.1.2 of RFC 7540)
+MUST NOT exceed the value of the SETTINGS_MAX_CONCURRENT_STREAMS parameter. A
+server that receives such a PRIORITY_UPDATE MUST respond with a connection error
+of type PROTOCOL_ERROR.
+
+When the PRIORITY_UPDATE frame applies to a push stream, a client SHOULD provide
+a Prioritized Stream ID that refers to a stream in the "reserved (remote)",
+"half-closed (local)", or "idle" state. Frames where the Prioritized Stream ID
```

Comment:
```suggestion
a Prioritized Stream ID that refers to a stream in the "reserved (remote)" or
"half-closed (local)" state. Frames where the Prioritized Stream ID
```
You can't prioritize a push in the idle state, because it hasn't been promised yet; you have no idea what it is.  Further, the reason there's not a limit on prioritizing pushed streams is because it's bound by the number of promises the server has issued.

It's not guaranteed detectable (because the PRIORITY_UPDATE could cross the PUSH_PROMISE on the network), but receiving prioritization for a push which is still "idle" should probably be an error.  That implies the client MUST NOT send PRIORITY_UPDATE for a server-initiated idle stream.

Reply:
I added that advice to the text, thanks.